Optik/Classes/AlbumViewController.swift

```diff
@@ -226,12 +280,18 @@ internal final class AlbumViewController: UIViewController {
                 return nil
             }
             
-            return ImageViewController(image: images[index], index: index)
+            pageControl?.numberOfPages = images.count
+            
+            let imageViewController =  ImageViewController(image: images[index], index: index)
+            imageViewController.likeButton?.addTarget(self, action: #selector(AlbumViewController.didSelectImage(_:)), for: .touchUpInside)
+            return imageViewController
```

Comment:
You are also creating a new ViewController and returning it before the end of the function, which means you are changing the logic. Hence my question about why are you doing that, instead of relying on what was already in place?